src/sonic-ctrmgrd/ctrmgr/kube_commands.py

```diff
@@ -211,6 +217,68 @@ def _download_file(server, port, insecure):
     log_debug("{} downloaded".format(KUBE_ADMIN_CONF))
 
 
+def _gen_cli_kubeconf(server, port, insecure):
+    """generate identity which can help authenticate and 
+       authorization to k8s cluster
+    """
+    client_kubeconfig_template = """
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: {{ k8s_ca }}
+    server: https://{{ vip }}:{{ port }}
+  name: kubernetes
+contexts:
+- context:
+    cluster: kubernetes
+    user: user
+  name: user@kubernetes
+current-context: user@kubernetes
+kind: Config
+preferences: {}
+users:
+- name: user
+  user:
+    client-certificate-data: {{ ame_crt }}
+    client-key-data: {{ ame_key }}
+    """
+    if insecure:
+        r = requests.get(K8S_CA_URL.format(server, port), cert=(AME_CRT, AME_KEY), verify=False)
```

Comment:
Should we skip providing the cert parameter when we want to do request in insecure mode?

Because I found the request will still try to validate SSL certificate even we set the verify to false.

Reply:
Should not, can you describe more details?